subgraphs/uniswap-v2/src/Factory.ts

```diff
@@ -0,0 +1,61 @@
+// import { log } from '@graphprotocol/graph-ts'
+import { PairCreated } from './../generated/Factory/Factory'
+import { DexAmmProtocol, LiquidityPool, Bundle } from './../generated/schema'
+import { Pair as PairTemplate } from '../generated/templates'
+import { getOrCreateToken, getOrCreateRewardToken } from './common/tokens'
+import {
+  FACTORY_ADDRESS,
+  BIGDECIMAL_ZERO
+} from './common/constants'
+
+export function handleNewPair(event: PairCreated): void {
+
+  let protocol = DexAmmProtocol.load(FACTORY_ADDRESS)
+
+  if (protocol === null) {
+    protocol = new DexAmmProtocol(FACTORY_ADDRESS)
+    protocol.name = "Uniswap v3"
+    protocol.slug = "uniswap-v3"
+    protocol.network = "ETHEREUM"
+    protocol.type = "EXCHANGE"
+
+    // create new bundle
+    let bundle = new Bundle('1')
+    bundle.ethPrice = BIGDECIMAL_ZERO
+    bundle.save()
+  }  
+
+  // create the tokens
+  let token0 = getOrCreateToken(event.params.token0)
+  let token1 = getOrCreateToken(event.params.token1)
+  let rewardToken0 = getOrCreateRewardToken(event.params.token0, "DEPOSIT")
```

Comment:
Use named constant for "DEPOSIT" as shown here: https://github.com/messari/subgraphs/blob/40d5822bf6dbdbcae185a7ccdd77b526c9dc5cbd/subgraphs/yearn-v2/src/common/constants.ts#L38

Reply:
Done!